homeassistant/components/statistics/sensor.py

```diff
@@ -292,31 +297,47 @@ async def async_stats_sensor_startup(_: HomeAssistant) -> None:
     def _add_state_to_queue(self, new_state: State) -> None:
         """Add the state to the queue."""
         self._available = new_state.state != STATE_UNAVAILABLE
-        if new_state.state == STATE_UNAVAILABLE:
+
+        if self._source_attribute:
+            new_state_value = new_state.attributes.get(self._source_attribute)
+            # TODO for review
+            # if self.states and new_state_value == self.states[-1]:
+            #     _LOGGER.debug(
+            #         "%s: Skipping state update for unchanged attribute value",
+            #         self.entity_id
+            #     )
+            #     return
```

Comment:
@emontnemery I wasn't sure how to handle non-changing values, which can happen for attributes. The commented out code would ignore those, analogue to the behavior with the entity state. This would in turn misbehave in the case of an `force_update` sensor.

Any steer would be great.